lib/spans/span-event.js

```diff
@@ -84,9 +84,21 @@ class SpanEvent {
    * @return {SpanEvent} The constructed event.
    */
   static fromSegment(segment, parentId = null, isRoot = false) {
+    const spanContext = segment.getSpanContext()
+
+    // Since segments already hold span agent attributes and we want to leverage
+    // filtering, we add to the segment attributes prior to processing.
+    if (spanContext.hasError && !segment.transaction.hasIgnoredErrorStatusCode()) {
+      const details = spanContext.errorDetails
+      segment.addSpanAttribute('error.message', details.message)
+      segment.addSpanAttribute('error.class', details.type)
+      if (details.expected) {
+        segment.addSpanAttribute('error.expected', details.expected)
+      }
+    }
+
```

Comment:
Don't particularly love this... but since we already are attaching span agent attributes to segments it avoids introducing a bunch of extra stuff for now.

Reply:
Ah, ya, looks like we're adding span attributes to the segment right before getting the attributes off the segment for creating a span.